server/src/test-fast/java/com/thoughtworks/go/server/controller/AgentRegistrationControllerTest.java

```diff
@@ -383,6 +383,28 @@ public void shouldAutoRegisterElasticAgent() {
         verify(agentService).requestRegistration(ElasticAgentRuntimeInfo.fromServer(agentRuntimeInfo, elasticAgentId, elasticPluginId));
     }
 
+    @Test
+    public void shouldNotRelyOnAutoRegisterKeyForRegisteringElasticAgents() {
+        String uuid = "elastic-uuid";
+        String autoRegisterKey = "auto_register_key";
+
+        final ServerConfig serverConfig = mockedServerConfig("token-generation-key", autoRegisterKey);
+        final String token = token(uuid, serverConfig.getTokenGenerationKey());
+
+        when(agentService.isRegistered(uuid)).thenReturn(false);
+        when(goConfigService.serverConfig()).thenReturn(serverConfig);
+        when(agentService.createAgentUsername(uuid, request.getRemoteAddr(), "host")).thenReturn(new Username("some-agent-login-name"));
+        when(ephemeralAutoRegisterKeyService.validateAndRevoke(any())).thenReturn(false);
+
+        String elasticAgentId = "elastic-agent-id";
+        String elasticPluginId = "elastic-plugin-id";
+        AgentRuntimeInfo.fromServer(new Agent(uuid, "host", request.getRemoteAddr()), false, "location", 233232L, "osx");
```

Comment:
@maheshp Is this line necessary? Looking at its impl, it doesn't seem to have any special side effects that were obvious to me. It doesn't _appear_ to hit any critical paths under test. Its return value is not used.

Also, when commenting out, the test still passes.

Reply:
Yes, it's redundant. I have changed it.